File: OCP/OCP.java
public class InsuerancePremiumDiscountCalculator {

    public int calculatePremiumDiscountPercent(CustomerProfile customer)
 {
     if(customer.isLoyalCustomer()) {
         return 20; 
     } 
     return 0; 
 }

}


public interface CustomerProfile {
    public boolean isLoyalCustomer(); 
}


public class HealtInsuranceCustomerProfile implements CustomerProfile {

    @Override 
    public boolean isLoyalCustomer() {
        return true; //or false 
    }

}

public class VehicleInsuranceCustomerProfile implements CustomerProfile {

    @Override 
    public boolean isLoyalCustomer() {
        return true; //or false 
    }

}

File: OCP/OCP.swift
class InsurancePremiumDiscountCalculator {
    
    public func calculatePremiumDiscountPercent(customer: CustomerProfile) -> Int {
        if customer.isLoyalCustomer() {
            return 20
        }
        return 0
    }
}

public protocol CustomerProfile {
    func isLoyalCustomer() ->  Bool
}


class HealthInsuranceCustomerProfile: CustomerProfile {
    func isLoyalCustomer() -> Bool {
        return true
    }
}

class VehicleInsuranceCustomerProfile: CustomerProfile {
    func isLoyalCustomer() -> Bool {
        return true
    }
}


class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        
        
        let discountRate = InsurancePremiumDiscountCalculator().calculatePremiumDiscountPercent(customer: HealthInsuranceCustomerProfile())
        print(discountRate)
        
    }


}

